### ISSUE TITLE
Many string changes and other customizations don't migrate

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/molecule-polarity#156]
**Device**
Samsung
**OS**
Win 11
**Browser**
Chrome
**Problem Description**
For https://github.com/phetsims/qa/issues/952
Several strings that can be changed in 1.2 do not stay changed in 1.3 when the file is migrated. Examples include the A and B on the molecules and the word Electronegativity. Nothing seemed to appear in the console indicating a problem that I could tell.
**Visuals**
![textchange2](https://github.com/phetsims/molecule-polarity/assets/41024075/7c3c6d03-a836-4673-9936-25a456d40cd9)
![notcarried](https://github.com/phetsims/molecule-polarity/assets/41024075/19b58ac5-e0b6-4cb3-9af5-018f8241ec20)

**Comment by KatieWoe** [comment]
I've also found this with at least one non-string element. `moleculePolarity.threeAtomsScreen.view.controlPanel.separator1.visibleProperty`
![seperatorgone](https://github.com/phetsims/molecule-polarity/assets/41024075/895220b7-fe25-471b-bc07-64d6c5e63e78)

**Comment by pixelzoom** [comment]
In the case of "Electronegativity", that tells me that this migration rule (in molecule-polarity-migration-rules.ts) is not working:

```ts
    new ConvertToStringsStateForLocale( /moleculePolarity\.\w+Screen\.view\.electronegativityPanels\.atom\wElectronegativityPanel\.subtitleText\.textProperty/,
      'moleculePolarity.general.model.strings.moleculePolarity.electronegativityStringProperty', stringsStatePhetioID ),
```

The phetioID in the previous version (1.2) is:
`moleculePolarity.twoAtomsScreen.view.electronegativityPanels.atomAElectronegativityPanel.subtitleText`

So the regex looks OK to me. But is `ConvertToStringsStateForLocale` handling the regex properly? And if not, why is there no error in the Migration Wrapper?

**Comment by pixelzoom** [comment]
Adding a similar ruler for a specific tandem, like this, does not fix the problem:

```diff
+   new ConvertToStringsStateForLocale( 'moleculePolarity.twoAtomsScreen.view.electronegativityPanels.atomAElectronegativityPanel.subtitleText.textProperty',
      'moleculePolarity.general.model.strings.moleculePolarity.electronegativityStringProperty', stringsStatePhetioID ),
    new ConvertToStringsStateForLocale( /moleculePolarity\.\w+Screen\.view\.electronegativityPanels\.atom\wElectronegativityPanel\.subtitleText\.textProperty/,
      'moleculePolarity.general.model.strings.moleculePolarity.electronegativityStringProperty', stringsStatePhetioID ),
```

So I'm guessing that the string customization gets set back to "Electronegativity" by subsequent applications of the rule to places where `subtitleText.textProperty` has not been customized.  If that's the case, then we can't map `textProperty` to `stringProperty`, and will need to change these rules to `TandemFragmentDelete`.  

I'll need to consult with @zepumph.

**Comment by pixelzoom** [comment]
In https://github.com/phetsims/molecule-polarity/issues/156#issuecomment-1579265846, @KatieWoe said:
> I've also found this with at least one non-string element. >`moleculePolarity.threeAtomsScreen.view.controlPanel.separator1.visibleProperty`

Separators are no longer instrumented, so those customizations are silently deleted.  This is reported in the console when the "Migrate now" button is pressed:

> TandemFragmentDelete: moleculePolarity.twoAtomsScreen.view.controlPanel.separator1.visibleProperty  

The relevant rule in molecule-polarity-migration-rules.ts are:

```ts
    // Separators (HSeparator) are no longer instrumented, because scenery layout automatically handles their visibility.
    // See https://github.com/phetsims/scenery/issues/1480
    new TandemFragmentDelete( 'moleculePolarity.twoAtomsScreen.view.controlPanel.separator1' ),
    new TandemFragmentDelete( 'moleculePolarity.twoAtomsScreen.view.controlPanel.separator2' ),
    new TandemFragmentDelete( 'moleculePolarity.threeAtomsScreen.view.controlPanel.separator1' ),
```

This demonstrates a problem that has been bothering me with migration.  The console output is super long, super noisy, and unlikely to be helpful to PhET-iO clients.  @zepumph @samreid FYI.

**Comment by pixelzoom** [comment]
@zepumph and I discussed this issue via Zoom.

For the migation problems that are due to many-to-one customizations, I created https://github.com/phetsims/phet-io-wrappers/issues/521.

For the problem of how QA and PhET-iO clients know what is supposed to migrate, what didn't migrate, etc. -- we punted for now, because we don't know how to proceed.  If the confusion continues to result in creation of GitHub issues, then we will revisit. 

So... This issue is on hold until https://github.com/phetsims/phet-io-wrappers/issues/521 is addressed.
